buf.gen.yaml

```diff
@@ -1,14 +1,14 @@
 version: v1
 plugins:
-  - remote: buf.build/protocolbuffers/plugins/cpp:v3.20.0-1
+  - plugin: buf.build/protocolbuffers/cpp:v23.0
```

Comment:
Newer versions of the `buf` CLI use the `plugin` key for both local and remote plugin references.